texar/modules/classifiers/xlnet_classifier.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+XLNet classifiers.
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import tensorflow as tf
+from texar import global_mode
+from texar.core import layers
+from texar.modules.classifiers.classifier_base import ClassifierBase
+from texar.modules import XLNetEncoder
+from texar.utils import utils
+from texar.hyperparams import HParams
+
+# pylint: disable=too-many-arguments, invalid-name, no-member,
+# pylint: disable=too-many-branches, too-many-locals, too-many-statements
+
+__all__ = [
+    "XLNetClassifier"
+]
+
+
+class XLNetClassifier(ClassifierBase):
```

Comment:
The pytorch version has a `param_group` method https://github.com/asyml/texar-pytorch/blob/master/texar/modules/classifiers/xlnet_classifier.py#L173

What's the counterpart here?

Reply:
I have implemented `param_groups` in a similar way like `texar-pytorch` in f578727. In tensorflow, to apply gradients, we call `optimizer.apply_gradients()` with a tuple containing pairs of `variable` and its corresponding gradient. I think maybe it would be better if `param_groups` returns a dict mapping variables to its learning rate, so that it becomes easier for the user to apply the learning rates to the corresponding gradient values. Let me know what you think. For the purpose of consistency however, we can keep it similar pytorch version.

Reply:
"`param_groups` returns a dict" --- this is better. 

Have we implemented the XLNet example as in PyTorch that demonstrates the use of the modules?

Reply:
Okay, I will change `param_groups` to return a dict mapping a `tf.Variable` to its learning rate. I have not implemented the examples, since I needed `XLNetEncoder` to be ready to build the examples.